x/launch/keeper/genesis_validator.go

```diff
@@ -54,3 +56,40 @@ func (k Keeper) GetAllGenesisValidator(ctx sdk.Context) (list []types.GenesisVal
 
 	return
 }
+
+// GetAllGenesisValidatorByLaunchID returns all genesisValidator by a launch id
+func (k Keeper) GetAllGenesisValidatorByLaunchID(ctx sdk.Context, launchID uint64) (list []types.GenesisValidator) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.GenesisValidatorAllKey(launchID))
+	iterator := sdk.KVStorePrefixIterator(store, []byte{})
+
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		var val types.GenesisValidator
+		k.cdc.MustUnmarshal(iterator.Value(), &val)
+		list = append(list, val)
+	}
+
+	return
+}
+
+// GetGenesisValidatorByConsPubKey returns the genesisValidator by consensus address
+func (k Keeper) GetGenesisValidatorByConsPubKey(
+	ctx sdk.Context,
+	launchID uint64,
+	consPubKey []byte,
+) (types.GenesisValidator, bool) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.GenesisValidatorAllKey(launchID))
+	iterator := sdk.KVStorePrefixIterator(store, []byte{})
+
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		var val types.GenesisValidator
+		k.cdc.MustUnmarshal(iterator.Value(), &val)
+		if bytes.Compare(val.ConsPubKey, consPubKey) == 0 {
+			return val, true
+		}
+	}
```

Comment:
I think we should have a single method that returns a map `map[conspubkey]GenesisValidator` and the total self delegation so we can do a single iteration of the validator set in `CheckValidatorSet`

Reply:
Mmh what do you think about it then? I was meaning doing a single iteration in validators `for ; iterator.Valid(); iterator.Next()` to get the total self delegation and map all validators by their cons pub key only once

Reply:
ohh sorry, I understand wrong. I agree. I also removed the `GetAllGenesisValidatorByLaunchID` so we can avoid duplicated interaction too

Reply:
done on https://github.com/tendermint/spn/pull/473/commits/f1ea06df25fe74578412ce2554ce6e97d34702aa